### PR TITLE
add UI page for notes list

### PR DESCRIPTION
**Pull request by mohit-s96**
## current progress screenshot
notes-ui-page current progress

## additional note

there is a vscode issue that causes it to not recognize the `@bigbinary/neetoui` package causing red squiggles all over the project.

I have temporarily suppressed it with `@ts-ignore`

### CODE REVIEW

**Comment by yedhink** [comment]
>there is a vscode issue that causes it to not recognize the @bigbinary/neetoui package causing red squiggles all over the project

Ignore this issue. It cannot be fixed in the near future.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        </Typography>
+        <Dropdown icon={MenuVertical} buttonStyle="text" position="auto-end">
+          <li className="my-1">
+            <span className="w-10/12 p-1">Edit</span>
+          </li>
+          <li className="my-1">
+            <span className="w-10/12 p-1">Delete</span>
+          </li>
+        </Dropdown>
+        {/* <Button icon={MenuVertical} style="text" /> */}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:27]: What's the point of keeping a comment and pushing the code or pertinently merge that code to master branch?

Comment coded should be removed before you even make a commit.

Self-Review is a must process in BigBinary. Whatever you are committing, you are accountable for that code. So make sure the code is clean and working and of course all Rails test cases are passing, before asking the reviewer to review.

Moving forward please keep this in mind.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        <div className="neeto-ui-border-gray-200 border-b-2 p-1"></div>
+      </div>
+    </div>
+    <div className="flex justify-between p-4">
+      {tags.map(({ tag }) => (
+        <Tag label={tag} key={tag} />
+      ))}
+      <div className="neeto-ui-text-gray-600 flex items-center gap-3">
+        <Clock size={18} />
+        <span>Created {createdAt} hours ago</span>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:42]: Use Typography from neetoUI.

File: app/javascript/src/components/Dashboard/Notes/Table.jsx [hunk 3]
```diff
@@ -1,27 +1,22 @@
...
-import { NOTES_TABLE_COLUMN_DATA } from "./constants";
+import NoteCard from "./NoteCard";
 import EditNotePane from "./Pane/Edit";
 
-const Table = ({ setSelectedNoteIds, notes = [], fetchNotes }) => {
+const Table = ({ notes = [], fetchNotes }) => {
   const [showEditNote, setShowEditNote] = useState(false);
-  const [selectedNote, setSelectedNote] = useState({});
-
+  const [selectedNote] = useState({});
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Table.jsx:8]: This is an anti-pattern. Why have you declared a state without a setter? If we don't need to set this state and rerender UI, then what's the point of making this a state?

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import React, { useState, useEffect } from "react";
 
+// @ts-ignore
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:3]: Remove these. These red squiggly lines will go away in future after we add relevant changes in neetoUI.

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 5]
```diff
@@ -57,23 +62,12 @@ const Notes = () => {
...
-          />
-          <Table
-            setSelectedNoteIds={setSelectedNoteIds}
-            notes={notes}
-            fetchNotes={fetchNotes}
-          />
-        </>
+        <Table
+          // @ts-ignore
+          setSelectedNoteIds={setSelectedNoteIds}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:67]: Why have you added ts-ignore here?

**Review by yedhink — COMMENTED** [review]
@mohit-s96 Few comments. Please carefully go through them and lemme know if you have any further queries.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx (hunk 1 above)
Comment (mohit-s96) [reply app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:27]: I must've missed it. Would keep it in mind further on.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx (hunk 2 above)
Comment (mohit-s96) [reply app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:42]: Would do.

File: app/javascript/src/components/Dashboard/Notes/Table.jsx (hunk 3 above)
Comment (mohit-s96) [reply app/javascript/src/components/Dashboard/Notes/Table.jsx:8]: Having the unused variable (`setSelectedNote`) was causing a pre-commit hook to fail so I just removed it. But as you pointed out, this isn't how it's done and I should've just made it a simple variable. Would keep it in mind.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 4 above)
Comment (mohit-s96) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:3]: Will do.

**Comment by mohit-s96** [comment]
Thanks for the review Yedhin. I have resolved the issues you pointed out. I'll create a new issue referencing this conversation and submit a PR.

**Pull request merged**